From 58f09e08b34b9f8769cc7516845d82cf3e9c25d9 Mon Sep 17 00:00:00 2001
From: William Wu <wulf@rock-chips.com>
Date: Thu, 8 Dec 2016 11:00:45 +0800
Subject: [PATCH] phy: rockchip-inno-usb2: set floating charger type as DCP

Floating charger is a kind of special USB charger that
dp/dm not shorted, which is incompatible with BC1.2 spec.

However, we need to support this floating charger, and
consider that the max charge current of floating charger
is the same as standard USB charger (dp/dm is shorted),
so we set the charger type as DCP for floating charger.

Change-Id: Ifaca7269a3d4660ac095c59776d7e935fe6126df
Signed-off-by: William Wu <wulf@rock-chips.com>
---
 drivers/phy/phy-rockchip-inno-usb2.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/phy/phy-rockchip-inno-usb2.c b/drivers/phy/phy-rockchip-inno-usb2.c
index 16e50f3b9d04..f5055f19b557 100644
--- a/drivers/phy/phy-rockchip-inno-usb2.c
+++ b/drivers/phy/phy-rockchip-inno-usb2.c
@@ -648,7 +648,7 @@ static void rockchip_usb2phy_otg_sm_work(struct work_struct *work)
 				case POWER_SUPPLY_TYPE_USB_FLOATING:
 					dev_dbg(&rport->phy->dev,
 						"floating cable is connecetd\n");
-					cable = EXTCON_CHG_USB_SLOW;
+					cable = EXTCON_CHG_USB_DCP;
 					rockchip_usb2phy_power_off(rport->phy);
 					sch_work = true;
 					break;
-- 
2.35.3

